lib/blockscore/base.rb

```diff
@@ -66,18 +67,20 @@ def capture_attributes(source)
       @attributes = source.attributes
     end
 
-    def add_accessor(symbol, *_args)
+    def add_accessor(symbol)
       singleton_class.instance_eval do
         define_method(symbol) do
-          wrap_attribute(attributes[symbol])
+          wrap_attribute(attributes.fetch(symbol))
         end
       end
     end
 
-    def add_setter(symbol, *_args)
+    def add_setter(symbol)
+      attr = symbol.to_s.chop.to_sym
+      assert_mutable(attr)
       singleton_class.instance_eval do
```

Comment:
What you are proposing sounds like you would have to know what attributes are permitted.